File: swTask7/bonus.py
import requests

# Настройки
api_token = 'y0_AgAAAABdx8V8AAr6sgAAAAD0lRNoGTu4N5UlS82mwTRCNB7t4tHHwLE'  # Замените на ваш токен
queue_id = 'TEAMCITYBUILDFA'    # ID созданной очереди
owner_id = 'aje5onvp7suh6hbr1jhh'    # ID владельца очереди

# URL для создания задачи
url = 'https://tracker.yandex.ru/createTicket?queue=TEAMCITYBUILDFA&_form=false'

headers = {
    'Authorization': f'OAuth {api_token}',
    'X-Org-ID': 'bpfbmihbfa98v3kflolj'  # ID вашей организации в Yandex Tracker
}

# Данные для создания задачи
data = {
    'queue': queue_id,
    'summary': 'Build Failure in TeamCity',
    'description': 'A build has failed in TeamCity. Please investigate.',
    'assignee': owner_id
}

# Отправка запроса на создание задачи
response = requests.post(url, headers=headers, json=data)

if response.status_code == 201:
    print('Task created successfully')
else:
    print('Failed to create task')

File: swTask7/bonus.cs
using System.Text;
using System.Text.Json;


public static class BuildTask
{
    public static readonly HttpClient httpClient = new HttpClient() { BaseAddress = new Uri("https://jsonplaceholder.typicode.com"), };
    public static async void Task(HttpClient client)
    {
        string host = "https://api.tracker.yandex.net";
        client.BaseAddress = new Uri(host);
        string direct = "/v2/issues";
        string token = "y0_AgAAAABdx8V8AAr6sgAAAAD0lRNoGTu4N5UlS82mwTRCNB7t4tHHwLE";
        client.DefaultRequestHeaders.Add("Authorizaton", $"OAuth {token}");

        string orgID = "bpfbmihbfa98v3kflolj";
        client.DefaultRequestHeaders.Add("X-Cloud-Org-ID", orgID );

        string[] arr = { "protsuk.yuriy", "iwhanter" };

        StringContent json = new(JsonSerializer.Serialize(new
        {
            queue = "TEAMCITYBUILDFA",
            summary = "И восстали машины из пепла ядерного огня",
            description = "машины не восстали",
            type = "task",
            priority = "critical",
            followers = arr,
            author = "dstrigin11",
            assignee = "dstrigin11",
        }),Encoding.UTF8, "application/json");
        var response = client.PostAsync(host + direct, json).Result; 
        if (response.IsSuccessStatusCode) 
        {
            string responseBody = response.Content.ReadAsStringAsync().Result;
            Console.WriteLine($"Response: {responseBody}");
        }
        else
        {
            Console.WriteLine($"error: {response.StatusCode} - {response.ReasonPhrase}");
        }
    }
}

